gst/nnstreamer/tensor_if/gsttensorif.c

```diff
@@ -205,23 +236,53 @@ gst_tensor_if_class_init (GstTensorIfClass * klass)
  * initialize instance structure
  */
 static void
-gst_tensor_if_init (GstTensorIf * self)
+gst_tensor_if_init (GstTensorIf * tensor_if)
 {
-  /** @todo NYI: initialize values of self */
-  self->silent = TRUE;
+  tensor_if->silent = TRUE;
+  gst_tensors_config_init (&tensor_if->in_config);
+  gst_tensors_config_init (&tensor_if->out_config);
+
+  tensor_if->sinkpad = gst_pad_new_from_static_template (&sink_factory, "sink");
+  gst_element_add_pad (GST_ELEMENT_CAST (tensor_if), tensor_if->sinkpad);
+  gst_pad_set_chain_function (tensor_if->sinkpad,
+      GST_DEBUG_FUNCPTR (gst_tensor_if_chain));
+  gst_pad_set_event_function (tensor_if->sinkpad,
+      GST_DEBUG_FUNCPTR (gst_tensor_if_event));
+
+  tensor_if->num_srcpads = 0;
+  tensor_if->srcpads = NULL;
 }
 
 /**
- * @brief Function to finalize instance. (GST Standard)
+ * @brief dispose function for tensor if (gst element vmethod)
  */
 static void
-gst_tensor_if_finalize (GObject * object)
+gst_tensor_if_dispose (GObject * object)
 {
-  /* GstTensorIf *self = GST_TENSOR_IF (object); */
+  G_OBJECT_CLASS (parent_class)->dispose (object);
+}
 
-  /** @todo NYI: finialize (free-up) everything in self */
+/**
+ * @brief Convert GValue to GList according to delimiters
+ */
+static GList *
+gst_tensor_if_set_property_glist (const GValue * value, GList * prop_list,
+    const gchar * delimiters)
+{
+  gint i;
+  gint64 val;
+  const gchar *param = g_value_get_string (value);
+  gchar **strv = g_strsplit_set (param, delimiters, -1);
+  gint num = g_strv_length (strv);
+  prop_list = NULL;
+
+  for (i = 0; i < num; i++) {
+    val = g_ascii_strtoll (strv[i], NULL, 10);
```

Comment:
How about checking any possible errors? You may use `errno` for this purpose.

Reply:
Thanks for your suggestion. I updated.